cartridges/int_cloudinary/cartridge/scripts/jobs/steps/cldUpdateProductImagesForFeeds.js

```diff
@@ -103,28 +103,30 @@ function writeProductFileContent(cloudinaryUrlStreamWriter, productSearchHitsItr
                     cloudinaryUrlStreamWriter.writeStartElement('image-group');
                     cloudinaryUrlStreamWriter.writeAttribute('view-type', params.viewType);
 
-                    for (var j = 0; j < imgAssetsSorted.resources.length; j++) {
+                    for (let imgResource of imgAssetsSorted.resources) {
                         cloudinaryUrlStreamWriter.writeStartElement('image');
-                        cloudinaryUrlStreamWriter.writeAttribute('path', imgAssetsSorted.resources[j].public_id);
+                        cloudinaryUrlStreamWriter.writeAttribute('path', imgResource.public_id);
                         cloudinaryUrlStreamWriter.writeEndElement();
                         cloudinaryUrlStreamWriter.writeCharacters('\n');
                     }
                     cloudinaryUrlStreamWriter.writeEndElement();
                     cloudinaryUrlStreamWriter.writeCharacters('\n');
 
                     if (!product.variants.empty) {
```

Comment:
Do we really need this condition?

Reply:
yeah, I think so, because on very next its making variant collection to iterator, if variants are empty it will get error while making it iterator

Reply:
It will return an iterator with hasNext() evaluating to false. Please remove the condition

Reply:
changes are done